admin/admin_token.go

```diff
@@ -0,0 +1,64 @@
+package admin
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+)
+
+/**
+ * GenerateAdminToken generates a new admin token for an election working for the current month
+ * Token is generated using HMAC-SHA256 with a secret key so that ensures that the token
+ * cannot be easily guessed or tampered with without knowledge of the key.
+ * Token will be valid for the "current" month (UTC) and the election ID.
+ */
+func GenerateAdminToken(electionId types.HexBytes) (string, error) {
+	secretKey := os.Getenv("CSP_ADMIN_TOKEN_SECRET")
+
+	currentTime := time.Now()
+	dateString := currentTime.Format("200601")
+
+	message := fmt.Sprintf("%s|%s", electionId.String(), dateString)
+
+	hmacHasher := hmac.New(sha256.New, []byte(secretKey))
+	_, err := hmacHasher.Write([]byte(message))
+	if err != nil {
+		return "", err
+	}
+	hash := hmacHasher.Sum(nil)
+	token := hex.EncodeToString(hash)
+	return token, nil
+}
+
+// ValidateAdminToken validates an admin token for an election
+func ValidateAdminToken(electionId types.HexBytes, adminToken string) (bool, error) {
+	generatedToken, err := GenerateAdminToken(electionId)
+	if err != nil {
+		return false, err
+	}
+
+	return adminToken == generatedToken, nil
+}
+
+// VerifySignatureForElection verifies that the signature is from the election creator
+func VerifySignatureForElection(electionId types.HexBytes, signature types.HexBytes, data types.HexBytes) (bool, error) {
+	address, err := ethereum.AddrFromSignature(data, signature)
+	if err != nil {
+		return false, err
+	}
+
+	// Verify the signer address is from the election creator
+	if strings.Contains(electionId.String(), strings.ToLower(address.String()[2:])) {
```

Comment:
Use this https://github.com/vocdoni/vocdoni-node/blob/main/vochain/processid/process_id.go#L29

```
p := processID.ProcessID{}
p.Unmarshal(electionId)
return bytes.Equal(p.Addr().Bytes(), address.Bytes()), nil
```
Note that if the function returns `false` because of verification failed, it might not be need to return an error (just return the error if there is an actual error).

Reply:
Yes, I think this Census origin check should be on "Creation"